packages/explorer/src/app/(home)/TableSelector.tsx

```diff
@@ -0,0 +1,53 @@
+import { Lock } from "lucide-react";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../../components/ui/Select";
+import { NON_EDITABLE_TABLES } from "../../consts";
+import { useWorldAddress } from "../../hooks/useWorldAddress";
+
+export function TableSelector({
+  value,
+  options,
+}: {
+  value: string | undefined;
+  options: string[];
+}) {
+  const worldAddress = useWorldAddress();
+  return (
+    <div className="py-4">
+      <Select
+        value={value}
+        onValueChange={(value: string) => {
+          const url = new URL(window.location.href);
+          const searchParams = new URLSearchParams(url.search);
+          searchParams.set("table", value);
+          window.history.pushState(
+            {},
+            "",
+            `${window.location.pathname}?${searchParams}`,
+          );
+        }}
+      >
+        <SelectTrigger>
+          <SelectValue placeholder="Select a table ..." />
+        </SelectTrigger>
+        <SelectContent>
+          {options?.map((option) => {
+            return (
+              <SelectItem key={option} value={option} className="font-mono">
+                {NON_EDITABLE_TABLES.includes(option) && (
+                  <Lock className="mr-2 inline-block opacity-70" size={14} />
+                )}
+                {option.replace(`${worldAddress}__`, "")}
```

Comment:
is there another way to derive the table name rather than string replacement?

Reply:
Alternatively, I could also just slice to show the last two parts of the name, namespace and table name e.g.

`0x8d8b6b8414e1e3dcfd4168561b9be6bd3bf6ec4b__world__balances` -> `world__balances`

I'm not too fond of string replacement either but table names are stored in such format by the indexer, and can't think of how else to do this?